test/test_nn.py

```diff
@@ -3483,6 +3483,23 @@ def test_multihead_attn_all_arguments3():
             test_multihead_attn_all_arguments2()  # Test MultiheadAttention with all the argument.
         test_multihead_attn_all_arguments3()  # Test MultiheadAttention with all the argument.
 
+    def test_multihead_attn_3d_attn_mask(self):
```

Comment:
What's the rationale behind this test?

Reply:
Given a 2d attention mask, we need to unsqueezed it for a 3d attention mask. Instead, I pass the equivalent 3d mask directly and compare the results.